nixos/modules/services/hardware/thermald.nix

```diff
@@ -4,27 +4,392 @@ with lib;
 
 let
   cfg = config.services.thermald;
+
+  # Prefixes every line in a string with the given prefix.
+  prefixLinesWith = prefix: s:
+    prefix + (builtins.replaceStrings ["\n"] ["\n${prefix}"] s);
+
+  indent = s: prefixLinesWith "  " s;
+
+  boolToInt = b: if b then 1 else 0;
+
+  mapXmlElement = f: cfg: attr:
+    let value = attrByPath [attr] null cfg;
+    in optionalString (value != null) "<${attr}>${f value}</${attr}>";
+
+  xmlBlock = name: children:
+    if children == []
+    then ""
+    else "<${name}>\n" + (indent (builtins.concatStringsSep "\n" children)) + "\n</${name}>";
+
+  xmlElement = cfg: attr: mapXmlElement (x: x) cfg attr;
+
+  platformConfig = with types; submodule {
+    options = {
+      Name = mkOption {
```

Comment:
Options are named according to camelCase, so all of the first letters should be lower cased instead (except UUID).

Reply:
Would we keep UUID as is? Or should it be `uuid`?

Reply:
Oh actually a couple options seem to be using `uuid`, so that's probably better